File: src/Sample.Invoker/Models/ArrayDesc.cs
using System;
using System.Runtime.InteropServices;

namespace Sample.Invoker.Models
{
    [StructLayout(LayoutKind.Sequential)]
    public readonly struct ArrayDesc
    {
        public ArrayDesc(IntPtr numbers, int length)
            : this()
        {
            this.Numbers = numbers;
            this.Length = length;
        }

        public readonly IntPtr Numbers;
        public readonly int Length;
    }
}

File: src/Sample.NativeLibrary/array_desc.cpp
#include <iostream>
#include "array_desc.h"
using namespace std;

ArrayDesc::ArrayDesc() = default;

ArrayDesc::ArrayDesc(const int* numbers, const int length)
{
	this->Numbers = numbers;
	this->Length = length;
}

ArrayDesc::~ArrayDesc()
{
	delete[] this->Numbers;
	std::cout << "C++ ArrayDesc析构函数调用" << std::endl;
}
